src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -318,46 +332,56 @@ func isCsrApproved(csr *certificatesv1.CertificateSigningRequest) bool {
 	return false
 }
 
-func (c controller) PostInstallConfigs(wg *sync.WaitGroup, status *ControllerStatus) {
-	defer wg.Done()
-	for {
-		time.Sleep(GeneralWaitInterval)
-		ctx := utils.GenerateRequestContext()
+func (c controller) PostInstallConfigs(ctx context.Context, wg *sync.WaitGroup, status *ControllerStatus) {
+	defer func() {
+		c.log.Infof("Finished PostInstallConfigs")
+		wg.Done()
+	}()
+	err := utils.WaitForPredicateWithContext(ctx, LongWaitTimeout, GeneralWaitInterval, func() bool {
+		ctxReq := utils.GenerateRequestContext()
 		cluster, err := c.ic.GetCluster(ctx)
 		if err != nil {
-			utils.RequestIDLogger(ctx, c.log).WithError(err).Errorf("Failed to get cluster %s from assisted-service", c.ClusterID)
-			continue
+			utils.RequestIDLogger(ctxReq, c.log).WithError(err).Errorf("Failed to get cluster %s from assisted-service", c.ClusterID)
+			return false
 		}
 		// waiting till cluster will be installed(3 masters must be installed)
 		if *cluster.Status != models.ClusterStatusFinalizing {
-			continue
+			return false
 		}
-		break
+		return true
+	})
+	if err != nil {
+		return
 	}
 
 	errMessage := ""
-	err := c.postInstallConfigs()
+	err = c.postInstallConfigs(ctx)
+	// context was cancelled
+	// no reason to set error
+	if ctx.Err() != nil {
```

Comment:
This distinction works only if we do not use context with Timeout but the existing predicate utility function. So it's need to be documented